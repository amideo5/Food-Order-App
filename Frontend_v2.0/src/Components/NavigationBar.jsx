import React from "react";
import { Navbar, Nav } from "react-bootstrap";
import Container from "react-bootstrap/Container";

const NavigationBar = (props) => {
  const handleLogout = () => {
    localStorage.removeItem("token");
    localStorage.removeItem("email");
    alert("Logged Out");
  };

  return (
    <Navbar
      className="m-0 py-3"
      collapseOnSelect
      expand="lg"
      bg="primary"
      variant="dark"
    >
      <Container fluid>
        <Navbar.Brand href="/welcome">NEX Food Ordering</Navbar.Brand>
        <Navbar.Toggle aria-controls="responsive-navbar-nav" />
        <Navbar.Collapse id="responsive-navbar-nav">
          <Nav className="me-auto">
            <Nav.Link></Nav.Link>
            <Nav.Link href="/welcome">Food List</Nav.Link>
            <Nav.Link href="/orderDetails">Orders</Nav.Link>
          </Nav>
          <Nav>
            <Nav.Link onClick={handleLogout} href="/">
              Logout
            </Nav.Link>
          </Nav>
        </Navbar.Collapse>
      </Container>
    </Navbar>
  );
};

export default NavigationBar;
