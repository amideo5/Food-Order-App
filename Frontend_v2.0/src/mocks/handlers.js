// src/mocks/handlers.js
import { rest } from "msw";

export const handlers = [
  rest.post("http://localhost:8080/users/signin", (req, res, ctx) => {
    return res(
      // Respond with a 200 status code
      ctx.status(200),
      ctx.body("Successful")
    );
  }),

  rest.post("http://localhost:8080/users/createUser", (req, res, ctx) => {
    return res(
      // Respond with a 200 status code
      ctx.status(200),
      ctx.body("Registered")
    );
  }),

  rest.get("http://localhost:8080/users/name/*", async (req, res, ctx) => {
    return res(
      // Respond with a 200 status code
      ctx.status(200),
      ctx.body("Aryan")
    );
  }),

  rest.get("http://localhost:8080/orders/getOrder/*", (req, res, ctx) => {
    return res(
      // Respond with a 200 status code
      ctx.status(200),
      ctx.json({
        orderName: "Aryan",
        items: {
          orderItems: {
            rajma: "2",
            chicken: "3",
          },
        },
        cost: "2000",
      })
    );
  }),

  rest.put("http://localhost:8080/users/updateUser/*", (req, res, ctx) => {
    return res(ctx.status(200), ctx.body("User Updated"));
  }),
  
];
