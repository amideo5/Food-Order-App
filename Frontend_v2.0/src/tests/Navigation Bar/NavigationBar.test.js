import React from "react";
import NavigationBar from "../../Components/NavigationBar";
import { getByText, render,screen } from "@testing-library/react";
import { fireEvent, waitFor } from "@testing-library/react";
import { server } from "../../mocks/server";
import { rest } from "msw";


describe("Testing Navbar Component",()=>{

    test("Snapshot testing",()=>{
        const {container} = render(<NavigationBar/>);

        expect(container).toMatchSnapshot();
    });

    test("Checking component is rendering or not",()=>{
        render(<NavigationBar/>);
        localStorage.setItem("token","sdjs");
        
        const navText = screen.getAllByText("NEX Food Ordering");
        
        expect(navText[0]).toBeInTheDocument();
    });

    test("renders Add Food link correctly", () => {
        render(<NavigationBar />);
        const addFoodLinkElement = screen.getByText(/Food List/i);
        expect(addFoodLinkElement).toBeInTheDocument();
      });
    
      test("renders Update Food link correctly", () => {
        render(<NavigationBar />);
        const updateFoodLinkElement = screen.getByText(/Orders/i);
        expect(updateFoodLinkElement).toBeInTheDocument();
      });

});

describe("handleLogout function", () => {
    it("should remove token and email from local storage", async() => {
      render(<NavigationBar />);
      const props = {
        history: { push: jest.fn() },
      };
      const mockAlert = jest.fn();
      localStorage.setItem("token", "test-token");
      localStorage.setItem("email", "test-email");
      
      fireEvent.click(screen.getByText("Logout"));

          
  
      expect(localStorage.getItem("token")).toBeNull();
      expect(localStorage.getItem("email")).toBeNull();
    });
  });