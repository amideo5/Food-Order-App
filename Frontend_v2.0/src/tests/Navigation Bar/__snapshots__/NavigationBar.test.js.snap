// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Navbar Component Snapshot testing 1`] = `
<div>
  <nav
    class="m-0 py-3 navbar navbar-expand-lg navbar-dark bg-primary"
  >
    <div
      class="container-fluid"
    >
      <a
        class="navbar-brand"
        href="/welcome"
      >
        NEX Food Ordering
      </a>
      <button
        aria-controls="responsive-navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          class="me-auto navbar-nav"
        >
          <a
            class="nav-link"
            href="#"
            role="button"
            tabindex="0"
          />
          <a
            class="nav-link"
            data-rr-ui-event-key="/welcome"
            href="/welcome"
          >
            Food List
          </a>
          <a
            class="nav-link"
            data-rr-ui-event-key="/orderDetails"
            href="/orderDetails"
          >
            Orders
          </a>
        </div>
        <div
          class="navbar-nav"
        >
          <a
            class="nav-link"
            data-rr-ui-event-key="/"
            href="/"
          >
            Logout
          </a>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
